public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -54,6 +82,39 @@ import {
 } from './reducer';
 import { useAlertQueryRunner } from './useAlertQueryRunner';
 
+export function areQueriesTransformableToSimpleCondition(
+  dataQueries: Array<AlertQuery<AlertDataQuery | ExpressionQuery>>,
```

Comment:
A bit confused about this type here; we have a separate input arg for expression queries so why are dataQueries `AlertDataQuery | ExpressionQuery` ? 🤔

Reply:
This is because of this [filter](https://github.com/grafana/grafana/blob/4f21ecf982c91959672e148f87664eb9b43aef48/public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx#L113) where we filter out expressions. So, we know that they are only dataQueries, but Typescript doesn't know. This dataQueries is the same used for the [QueryEditor](https://github.com/grafana/grafana/blob/4f21ecf982c91959672e148f87664eb9b43aef48/public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx#L446)